test/envtest/kongpluginbinding_managed_test.go

```diff
@@ -117,113 +72,326 @@ func TestKongPluginBindingManaged(t *testing.T) {
 	require.NoError(t, clientNamespaced.Create(ctx, rateLimitingkongPlugin))
 	t.Logf("deployed %s KongPlugin (%s) resource", client.ObjectKeyFromObject(rateLimitingkongPlugin), rateLimitingkongPlugin.PluginName)
 
-	wKongPluginBinding := setupWatch[configurationv1alpha1.KongPluginBindingList](t, ctx, clientWithWatch, client.InNamespace(ns.Name))
-	wKongPlugin := setupWatch[configurationv1.KongPluginList](t, ctx, clientWithWatch, client.InNamespace(ns.Name))
-	kongService := deployKongService(t, ctx, clientNamespaced,
-		&configurationv1alpha1.KongService{
-			ObjectMeta: metav1.ObjectMeta{
-				Annotations: map[string]string{
-					consts.PluginsAnnotationKey: rateLimitingkongPlugin.Name,
+	t.Run("binding to KongService", func(t *testing.T) {
+		serviceID := uuid.NewString()
+
+		pluginID := uuid.NewString()
+		sdk.PluginSDK.EXPECT().
+			CreatePlugin(mock.Anything, cp.GetKonnectStatus().GetKonnectID(), mock.Anything).
+			Return(
+				&sdkkonnectops.CreatePluginResponse{
+					Plugin: &sdkkonnectcomp.Plugin{
+						ID: lo.ToPtr(pluginID),
+					},
 				},
+				nil,
+			)
+
+		wKongPluginBinding := setupWatch[configurationv1alpha1.KongPluginBindingList](t, ctx, clientWithWatch, client.InNamespace(ns.Name))
+		wKongPlugin := setupWatch[configurationv1.KongPluginList](t, ctx, clientWithWatch, client.InNamespace(ns.Name))
+		// wKongService := setupWatch[configurationv1alpha1.KongServiceList](t, ctx, clientWithWatch, client.InNamespace(ns.Name))
+
+		kongService := deployKongServiceAttachedToCP(t, ctx, clientNamespaced, cp,
+			WithAnnotation(consts.PluginsAnnotationKey, rateLimitingkongPlugin.Name),
+		)
+		t.Cleanup(func() {
+			require.NoError(t, clientNamespaced.Delete(ctx, kongService))
+		})
```

Comment:
Why do we care about cleaning up these objects (I'm thinking of any other object we clean up in this PR)? Do you think that's needed, given the environment is spun up for each test anyway and names are generated?

Reply:
I want to try retain as much isolation between subtests as possible and given that these objects (like `KongService`) can cause creation or update of `KongPluginBinding`s I'd prefer to have as little env overlap between tests as possible.

Reply:
I see. I pointed this out as this might be not obvious which objects we want to clean up and why. Do you mind leaving some comments explaining that?

Reply:
👍 

13bd1a3f484c311ed9424134cbc732c827f8c456 covers that.